src/relay/blockchain/events_informations.py

```diff
@@ -0,0 +1,129 @@
+    Assumes the trustline exists"""
+
+    balance_update_events = events_proxy.get_trustline_events(
+        currency_network_address, user, counterparty, event_name=BalanceUpdateEventType
+    )
+    trustline_update_events = events_proxy.get_trustline_events(
+        currency_network_address,
+        user,
+        counterparty,
+        event_name=TrustlineUpdateEventType,
+    )
+
+    return get_accrued_interests_from_events(
+        balance_update_events, trustline_update_events
+    )
+
+
+def get_accrued_interests_from_events(balance_update_events, trustline_update_events):
+    accrued_interests = []
+    for (pre_balance_event, post_balance_event) in toolz.itertoolz.sliding_window(
+        2, balance_update_events
+    ):
+        balance = balance_viewed_from_user(pre_balance_event)
+        interest_rate = get_interests_rates_of_trustline_for_user_before_timestamp(
+            trustline_update_events, balance, post_balance_event.timestamp
+        )
+        interest_value = calculate_interests(
+            balance,
+            interest_rate,
+            post_balance_event.timestamp - pre_balance_event.timestamp,
+        )
+        accrued_interests.append(
+            InterestAccrued(interest_value, interest_rate, post_balance_event.timestamp)
+        )
+
+    return accrued_interests
+
+
+def balance_viewed_from_user(balance_update_event):
+    if balance_update_event.direction == DIRECTION_SENT:
+        return balance_update_event.value
+    elif balance_update_event.direction == DIRECTION_RECEIVED:
+        return -balance_update_event.value
+    else:
+        raise RuntimeError("Unexpected balance update event")
+
+
+def get_interests_rates_of_trustline_for_user_before_timestamp(
+    trustline_update_events, balance, timestamp
+):
+    """Get the interest rate that would be used to apply interests at a certain timestamp"""
+
+    most_recent_event_before_timestamp = None
+    for event in sorted_events(trustline_update_events, reversed=True):
+        if event.timestamp < timestamp:
+            most_recent_event_before_timestamp = event
+            break
+    if most_recent_event_before_timestamp is None:
+        raise RuntimeError("No trustline update event found before given timestamp")
+
+    if most_recent_event_before_timestamp.direction == DIRECTION_SENT:
+        if balance >= 0:
+            return most_recent_event_before_timestamp.interest_rate_given
+        else:
+            return most_recent_event_before_timestamp.interest_rate_received
+    elif most_recent_event_before_timestamp.direction == DIRECTION_RECEIVED:
+        if balance >= 0:
+            return most_recent_event_before_timestamp.interest_rate_received
+        else:
+            return most_recent_event_before_timestamp.interest_rate_given
+    else:
+        raise RuntimeError("Unexpected trustline update event")
+
+
+def sorted_events(events, reversed=False):
```

Comment:
Please rename the second argument to `reverse`, so it used the same name as `sorted`.